egs/hkust/s5/local/nnet3/run_ivector_common.sh

```diff
@@ -0,0 +1,153 @@
+  fi
+
+  for datadir in train dev; do
+    utils/copy_data_dir.sh data/$datadir data/${datadir}_hires
+    if [ "$datadir" == "train" ]; then
+      dir=data/train_hires
+      cat $dir/wav.scp | python -c "
+import sys, os, subprocess, re, random
+scale_low = 1.0/8
+scale_high = 2.0
+for line in sys.stdin.readlines():
+  if len(line.strip()) == 0:
+    continue
+  print '{0} sox --vol {1} -t wav - -t wav - |'.format(line.strip(), random.uniform(scale_low, scale_high))
+"| sort -k1,1 -u  > $dir/wav.scp_scaled || exit 1;
+     mv $dir/wav.scp $dir/wav.scp_nonorm
+     mv $dir/wav.scp_scaled $dir/wav.scp
+    fi
+
+    steps/make_mfcc.sh --nj 70 --mfcc-config conf/mfcc_hires.conf \
+      --cmd "$train_cmd" data/${datadir}_hires exp/make_hires/$datadir $mfccdir || exit 1;
+    steps/compute_cmvn_stats.sh data/${datadir}_hires exp/make_hires/$datadir $mfccdir || exit 1;
+  done
+fi
+
+if [ $stage -le 2 ]; then
+  # Train a system just for its LDA+MLLT transform.  We use --num-iters 13
+  # because after we get the transform (12th iter is the last), any further
+  # training is pointless.
+  steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \
+    --realign-iters "" \
+    --splice-opts "--left-context=3 --right-context=3" \
+    5000 10000 data/train_hires data/lang \
+    ${gmm_dir}_ali exp/nnet3/tri5
+fi
+
+if [ $stage -le 3 ]; then
+  steps/online/nnet2/train_diag_ubm.sh --cmd "$train_cmd" --nj 30 \
+    --num-frames 700000 \
+    --num-threads $num_threads_ubm \
+    data/train_hires 512 exp/nnet3/tri5 exp/nnet3/diag_ubm
+fi
+
+if [ $stage -le 4 ]; then
+  # iVector extractors can in general be sensitive to the amount of data, but
+  # this one has a fairly small dim (defaults to 100)
+  steps/online/nnet2/train_ivector_extractor.sh --cmd "$train_cmd" --nj 10 \
+    data/train_hires exp/nnet3/diag_ubm exp/nnet3/extractor || exit 1;
+fi
+
+if [ $stage -le 5 ] && [ "$speed_perturb" == "true" ]; then
+  # Although the nnet will be trained by high resolution data,
+  # we still have to perturbe the normal data to get the alignment
+  # _sp stands for speed-perturbed
+  utils/perturb_data_dir_speed.sh 0.9 data/train data/temp1
+  utils/perturb_data_dir_speed.sh 1.0 data/train data/temp2
+  utils/perturb_data_dir_speed.sh 1.1 data/train data/temp3
+  utils/combine_data.sh --extra-files utt2uniq data/train_sp data/temp1 data/temp2 data/temp3
+  rm -r data/temp1 data/temp2 data/temp3
+
+  mfccdir=mfcc_perturbed
+  for x in train_sp; do
+    steps/make_mfcc.sh --cmd "$train_cmd" --nj 70 \
+      data/$x exp/make_mfcc/$x $mfccdir || exit 1;
+    steps/compute_cmvn_stats.sh data/$x exp/make_mfcc/$x $mfccdir || exit 1;
+  done
+  utils/fix_data_dir.sh data/train_sp
+
+  $align_script --nj 30 --cmd "$train_cmd" \
+    data/train_sp data/lang $gmm_dir ${gmm_dir}_sp_ali || exit 1
+
+  # Now perturb the high resolution daa
```

Comment:
Maybe this comment should be
# Now generate high resolution data from perturbed data